setuptools/command/build_ext.py

```diff
@@ -104,14 +104,20 @@ def copy_extensions_to_source(self):
                 self.write_stub(package_dir or os.curdir, ext, True)
 
     def get_ext_filename(self, fullname):
-        filename = _build_ext.get_ext_filename(self, fullname)
+        so_ext = os.getenv('SETUPTOOLS_EXT_SUFFIX')
+        if so_ext:
+            filename = os.path.join(*fullname.split('.')) + so_ext
+        else:
+            filename = _build_ext.get_ext_filename(self, fullname)
+            so_ext = get_config_var('EXT_SUFFIX')
```

Comment:
Have you considered adding a hook inside `get_config_var` or `get_ext_filename` that would address the need more fundamentally?

Reply:
Yeah, that's where I started. Obviously that only works when you also opt in to using the `distutils` that comes with `setuptools`, but that's fine.

The bigger issue is that the code later in this function tries to strip off the suffix later, which is not possible to do if `get_ext_filename` uses anything other than `get_condig_var('EXT_SUFFIX')` (at least without the same logic as what I added here anyway).

Adding the hook in `setuptools._distutils.sysconfig.get_config_var` would have made some sense. However, I assume at some point in the future that the patches for `build_ext` will eventually merge _somewhere_, and at some point `distutils.sysconfig` will be replaced with the standard library's `sysconfig` module. If the hook were in `_distutils`, there's a pretty good chance it would be forgotten later and the functionality would be lost. This way, there's no chance of it being forgotten in any refactorings.

Reply:
(And yes, the tests would detect the functionality being lost in a refactoring. But tracking down the root cause and handling it - probably just like this - would take up someone's day and I don't want to inflict that on them!)